pkg/plugin/plugin.go

```diff
@@ -488,8 +489,16 @@ func CmdAdd(args *skel.CmdArgs) error {
 	return types.PrintResult(result, netconf.CNIVersion)
 }
 
-func waitLinkUp(ovsDriver *ovsdb.OvsBridgeDriver, ofPortName string) error {
-	for i := 1; i <= linkstateCheckRetries; i++ {
+func waitLinkUp(ovsDriver *ovsdb.OvsBridgeDriver, ofPortName string, netConf *netConf) error {
```

Comment:
Please pass individual attributes to this function, not the whole netConf struct.

Passing the whole config to this helper function poisons scoping. Also, all the other functions called from CmdAdd are getting only selected attributes.

Finally, I would prefer if you kept netConf defaulting on a higher level. Otherwise, in theory, two helper functions using this value can use a different defaulting logic. But I won't fight for it in this PR.